taui/src/components/edit-profile.js

```diff
@@ -198,80 +243,138 @@ export default class EditProfile extends PureComponent<Props> {
+      }).then(response => {
+        if (response.error) {
+          console.error('Failed to create user')
+          console.error(response)
+          if (response.result === 'userExists') {
+            console.warn('Account already exists')
+            console.warn(response.user)
+            // Find the voucher number for the existing user account
+            const existingVoucherNumber = this.getUserDataVoucherNumber(response.user)
+            if (existingVoucherNumber) {
+              if (existingVoucherNumber === key) {
+                // Search should have found the existing profile, if there was any
+                // so assume this is newly created.
+                console.warn('Client account already exists with matching voucher number')
+                // Update state to show client account for matching email and voucher number
+                // has already been confirmed.
+                this.setState({clientAccountConfirmed: true})
+                resolve('exists')
+              } else {
+                console.error('Existing client account is for a different voucher number')
+                // Let counselor know that there is already a Cognito account for that e-mail
+                this.setState({errorMessage: message('Profile.CreateClientAccountExistsError', {
+                  voucher: existingVoucherNumber
+                })})
+                resolve('failed')
+              }
+            } else {
+              // No voucher number on existing user account. Is it the email of a counselor?
+              console.warn('No voucher number on existing profile. Is that a counselor email?')
+              this.setState({errorMessage: message('Profile.CreateClientAccountError')})
+            }
+          } else if (response.result === 'inviteNotResentVoucherMismatch') {
+            this.setState({errorMessage: message('Profile.CreateClientAccountExistsError', {
+              voucher: this.getUserDataVoucherNumber(response.user)
+            })})
+          } else {
+            console.error('Unrecognized error attempting to create user')
+            this.setState({errorMessage: message('Profile.CreateClientAccountError')})
+          }
+          resolve('failed')
+        } else {
+          console.log('User created (or invite resent)')
+          // TODO: show different message if invite resent?
+          // if (response.result && response.result === 'resendingInvite') {
+          this.setState({errorMessage: '', clientInviteSent: true})
+          resolve('created')
+        }
+      }).catch(error => {
+        // A 403 (as when user is not a counselor) will only return "Network Error"
+        console.error('API call to create user failed')
+        console.error(error)
+        this.setState({errorMessage: message('Profile.CreateClientAccountError')})
+        reject(error)
+      })
     })
   }
 
-  deleteProfile (key, event) {
+  deleteProfile (key: string, isCounselor: boolean, event) {
```

Comment:
One behavior I noticed (not sure if it's an issue, necessarily) is that once a profile is deleted, a counselor can re-create a profile for the same voucher number, but with a different email address. If the second email address signs in, then the result is two Cognito logins being associated with the same voucher number. It doesn't seem to cause any major harm -- both emails can log in and see the same profile info, which I guess would probably be the desired behavior in this case. However, the email address displayed in the profile will only ever be the most recent email address that was associated with the voucher, so you can log in with the "older" email, and the email displayed in the profile page will be the "newer" one.

I'm not sure whether this is actually a problem, since it seems like a fairly rare case, and both emails can still log in and see the voucher info, so it wouldn't allow someone to get locked out. This also enables counselors to change the email address associated with an account in case they make a mistake, so it might be a useful escape hatch. Do we have enough info from the client to know whether this matches what they wanted?

Reply:
I modified the `/profiles` endpoint to delete the old login account in the case that a client logs in without a profile, and another profile exists for their voucher number but a different email address.